grails-app/controllers/au/org/ala/volunteer/NewsItemController.groovy

```diff
@@ -0,0 +1,278 @@
+            log.debug("NewsItem has errors: ${newsItem.errors}")
+            render view: 'create', model: [newsItem: newsItem]
+            return
+        }
+
+        newsItem.save(flush: true, failOnError: true)
+
+        // optional thumbnail upload handling
+        if (request instanceof MultipartHttpServletRequest) {
+            MultipartFile f = ((MultipartHttpServletRequest) request).getFile('newsItemThumb')
+            if (f != null && f.size > 0) {
+                def allowedMimeTypes = ['image/jpeg', 'image/png']
+                if (!allowedMimeTypes.contains(f.getContentType())) {
+                    flash.message = "Image must be one of: ${allowedMimeTypes}"
+                } else {
+                    // Upload image
+                    def result = newsItemService.uploadImage(newsItem.id, f)
+                    if (!result) {
+                        flash.message = "Error uploading image"
+                        redirect(action: 'create', params: params)
+                        return
+                    } else {
+                        log.debug("Image uploaded successfully: ${result}")
+                    }
+                }
+            }
+        }
+
+        flash.message = message(code: 'default.created.message', args: [message(code: 'newsItem.label', default: 'NewsItem'), newsItem.title])
+        redirect(action: 'manage')
+    }
+
+    def edit(Long id) {
+        if (!userService.isSiteAdmin()) {
+            flash.message = "You do not have permission to view this page"
+            render(view: '/notPermitted')
+            return
+        }
+        def datePickerRanges = getDatePickerRanges()
+        render view: 'edit', model: [newsItem: NewsItem.get(id), defaultEndDate: datePickerRanges.endDateStr, defaultStartDate: datePickerRanges.startDateStr]
+    }
+
+    def clearImage(NewsItem newsItem) {
+        if (!userService.isSiteAdmin()) {
+            flash.message = "You do not have permission to view this page"
+            render(view: '/notPermitted')
+            return
+        }
+
+        if (newsItem) {
+            // Clear the image associated with the news item
+            def result = newsItemService.deleteImage(newsItem.id)
+            if (!result) {
+                flash.message = "Error removing image for news item with ID ${newsItem.id}"
+            } else {
+                log.debug("Image removed successfully for news item with ID ${newsItem.id}")
+                flash.message = "Image removed successfully."
+            }
+        }
+
+        redirect(action: 'edit', id: newsItem.id)
+    }
+
+    @Transactional
+    def update(NewsItem newsItem) {
+        if (!userService.isSiteAdmin()) {
+            flash.message = "You do not have permission to view this page"
+            render(view: '/notPermitted')
+            return
+        }
+        log.debug("Updating news item with params: ${params}")
+
+        if (params.dateExpiresPicker) {
```

Comment:
This looks like a duplicate of above (save vs update) - could refactor into a method or command object ?

Reply:
Thanks, good idea.